### ISSUE TITLE
Hide Key Metrics banner/widget area when SC and GA are in a Zero Data State

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6512]
## Feature Description

This requirement was split out of #6209 as that issue was getting [too large as per this comment](https://github.com/google/site-kit-wp/issues/6209#issuecomment-1400336625). 

Ideally, we do not want to show the Key Metrics Setup CTA widget on the dashboard if SC or Analytics are either in Gathering Data or Zero Data states. To avoid "flickers" where the widget is first shown and then hidden immediately after we've loaded our reports to check for gathering data / zero data, we created #5933 which "pre-loads" information regarding gathering data and this issue is supposed to tackle the same concern for zero data. 

However, since zero data is not really a plugin "state" but depends on the date range selected in the plugin, it is not possible to pre-load this information. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When loading the Site Kit dashboard, if Search Console and GA4 return "zero data", the Key Metrics Setup CTA widget should not be displayed. (The Key Metrics widget area should not be displayed at all).
* When on subsequent page loads, if Search Console and GA4 do NOT return "zero data", the Key Metrics Setup CTA should be displayed. 
* Once the dashboard is loaded, the Setup CTA should not pop up and cause movement on the page if the site "leaves" the zero data state. The Setup CTA should only be displayed on a page load (/reload).
* If in the [rare case](https://github.com/google/site-kit-wp/issues/6512#issuecomment-1434856240) that the site "enters" zero data state _after_ the Key Metrics widget has been setup, then nothing should happen, ie. the Key Metrics widget should still be displayed the way it was before the site entered zero data state.

## Implementation Brief

Notes for IB author:
* When loading the Site Kit dashboard, check if Search Console and GA4 return Zero Data for the date range selected. If they do NOT return zero data, set the `isWidgetHidden` flag in the Key Metrics setting to `false`. 
* Ensure the default value for the `isWidgetHidden` flag is set to `true`.
* Use the `isWidgetHidden` flag to show/hide the widgets registered to the Key Metrics widget area.

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by marrrmarrr** [comment]
Hey @jimmymadon, just one thing to clarify -- this would apply only to initial setup CTA for the KMW, correct? 

@aaemnnosttv and I discussed this question briefly as well last week, and there was also an option 3, namely show the KMW setup CTA the *following time* a user logs in after we detect a site is no longer in zero state. So this flow would be: 
- user logs in --> check if there are stats to show for the site
   - if no stats, do nothing
   - if there are stats, make a note that the site has "left" zero state
- next time the user logs in after the site "left" zero state, render the KMW CTA. 

Curious to hear everyone's thoughts on this option.

**Comment by marrrmarrr** [comment]
@jimmymadon I think the next page reload is also fine, as long as it's a "new view", so we avoid the jumpiness of elements shifting vertically. 
Regarding zero data, I think we currently call more than one state "zero data", so sometimes confusion arises. There are a few options:
1. recently created site, no visitors yet, so widgets show zeros
2. site which didn't have SC/GA accounts prior to setting up SK - will be in zero/gathering data state until stats are aggregated, then might either start showing stats or go into state (1) above if not enough visitors. 
3. site which for some reason lost all traffic -- eventually the dashboard would zero out. 

In case (3), it could happen that the already existing KMW shows all zeros. Admittedly this is a corner case.

**Comment by aaemnnosttv** [comment]
After discussing a bit with Felix today, it would probably be good to open a new issue to implement the persistent zero data handling as an extension of the related work we've done in #5933. I'll open an issue for that one and which this issue will depend on and be much simpler to implement.

**Comment by tofumatt** [comment]
@aaemnnosttv Not sure why I was assigned to this IB, I think I was just reviewing the ACs but I've unassigned myself—I don't think there's anything special here that needs me unless I'm mixed up on something 🙂

**Comment by marrrmarrr** [comment]
@jimmymadon if it's significant work to get this done, sounds good to postpone it.

**Comment by aaemnnosttv** [comment]
I don't believe this is relevant any more as we don't surface KM setup for sites that are gathering data.
